File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    public class UserParams : PaginationParams
    {
        
        public string CurrentUsername { get; set; }
        public string Gender { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 80;
        public string SearchChar { get; set; }
        public string OrderBy { get; set; } = "lastActive";

    }
}

File: client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams {
    gender: string;
    minAge = 18;
    maxAge = 80;
    pageNumber = 1;
    pageSize = 5;
    orderBy = 'lastActive';
    searchChar = '';

    constructor(user: User) {
        this.gender = user.gender;
    }
}